app/models/user.rb

```diff
@@ -0,0 +1,21 @@
+class User < ApplicationRecord
+  VALID_EMAIL_REGEX = Settings.user.validate.VALID_EMAIL_REGEX
+
+  validates :name, presence: true,
+    length: {maximum: Settings.user.validate.name_max}
+  validates :email, presence: true,
+    length: {maximum: Settings.user.validate.email_max},
+    format: {with: VALID_EMAIL_REGEX}, uniqueness: true
+  validates :password, presence: true,
+    length: {minimum: Settings.user.validate.pass_min}
+
+  has_secure_password
```

Comment:
hàm này có tác dụng gì nhỉ

Reply:
hàm has_secure_password được dùng để mã hóa và xác nhận bằng cách dùng mật khẩu bcrypt.
Lưu thuộc tính password_digest được băm vào cơ sở dữ liệu.
has_secure_password tự động thêm một phương thức xác thực vào các đối tượng tương ứng 
Xác định xem một mật khẩu đã cho có hợp lệ đối với một người dùng cụ thể hay không bằng cách tính toán và so sánh kết quả với password_digest trong cơ sở dữ liệu.